android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/translate/GeckoTranslationState.kt

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.engine.translate
+
+/**
+* The representation of the Gecko translations engine toolkit state.
+*
+* @property detectedLanguages Detected information about preferences and page information.
+* @property error If an error state occurred or an error was reported by toolkit.
+* @property isEngineReady If the toolkit translation engine is primed for use or will need to be loaded.
+* @property requestedTranslationPair The language pair to translate. Usually populated after first request.
+*/
+
+data class GeckoTranslationState(
+    val detectedLanguages: DetectedLanguages? = null,
+    val error: String? = null,
+    val isEngineReady: Boolean? = false,
+    val requestedTranslationPair: TranslationPair? = null,
```

Comment:
Could properties `DetectedLanguages` and `TranslationPair` could be part of the `GeckoTranslationState` class?

Reply:
They could, but I think we may lose some of the conceptual lifecycle if we go that route.

 Main reasoning for current implementation is that this state change on those items only happens through the `onTranslationStateChange`.  I would like to make it conceptually clear that those items all share the same lifecycle. Open to this change though, just want to make sure it is the direction we'd like to go!